### DISCUSSION TITLE
Type issue when initializing a `ref()` with an object that includes router types

### DISCUSSION THREAD

**Discussion by JammingBen** [vuejs/router#2373] [Help and Questions] [answered]
I'm currently facing an issue where TS complains about incorrect types when initializing a `ref()` with an object that includes a property of type `RouteLocation`. 

The code looks something like this:

```
type FooType = {
  name: string,
  to?: RouteLocation
}

const foo: FooType = { name: 'test' };
const bar: Ref<FooType> = ref(foo)
```

Now TS is complaining that `foo` isn't assignable to the type `FooType`. It seems to come down to this issue: `Types of property 'path' are incompatible. Type 'unknown' is not assignable to type 'undefined'`.

Also, I only encounter this error with `vue-router` versions >= `4.3.0`. Anyone got an idea why this is happening? 

Reproduction: [Vue SFC Playground](https://play.vuejs.org/#eNp9klFv0zAUhf+KFR7SSa0N2rSH0I0CGhIIASp7zIub3mTuEtuyb9qiKv+dayekGYM9VI3u/XJ8znFOSSOV5jufZIlqrHHITqxwIBHeW8s6VjrTsHTfQprrAQiLfsyFtJb3y1yPr83od5FrxnhjWo2z9BVh6UUyTwacDlv6wimLzAO2ltVSVzd5gj5PbsdzTmwN5Zw5KJ/4eDsFTIvw1RQSldFreZiCCxeWLvC5xl8W2Cdj7sP/DTsFd1o2kDGPTulqHgZo3mXPJHPdxXBGe2SlMdlUZtBIETymrKOjem4jHSlBuRzYW4Ipx4zep2KWog9PUamTPsyikZauwWjqJrrLhwVVkvV+w+ycK4zz5AHR+kyIVtvHihemEWdidcUv+WuxVR4nUw6+WWycOXh63pF6jB60VwSJLezRmNovpFX/O+IZuLrm1/yNqNVGkLpQegvHqB2kqb6OYqKnakpV/RWS9KyqwX23oe+nYWVdm8OXOEPXwmi0eIDi8R/znT/2ln84oHR7mIRD6SrAfn338xsc6XlcNmbb1kS/sFyDN3UbPPbYh1ZvyfaEi24/xzujD+re3x0RtP8TKhiNbUQ+XuTHF6Kf7V7yq7HF7jdbxD2D) (see `app.vue` file)

**Comment by posva** [comment]
You should pass the type param to `ref` directly: `const bar = ref<FooType>(foo)`

**Reply by JammingBen** [reply, marked as answer]
I thought about that and while the error does indeed disappear, it's but only because `bar` is actually not of type `FooType` in this case. So it becomes an issue when e.g. passing `bar` to a function that expects a parameter of type `FooType`: [Check here](https://play.vuejs.org/#eNp9kk9v00AQxb/KyhycSska1KoHk4QAKhIIAUp79GVjTxyn3j/aHSdBUb47s7up41LowbI189s3b976mEjRKL51SZ400miL7MhKCwLhozHsxNZWS5buOkgLdQZ8I5Z5JozhsVmo/tiInqtCMcal7hSO0jeEpVfJODnjNGzqStsYZA6wM6wVqp4VCboimfdzjmwJ6zGzsH7m4/0Q0B3Cd10KbLRaiv0QnFjftJ4vFP42wL5o/eDfM3b07pSQkDOHtlH12BdQf8hfSBbqFJbTyiFba50PZc4aKYLDlJ1oVORWwlKbnE/P8HxERymTJ6DS9xI3xIyMsEL2oldsNo/uPKdb4K2uI0Ong5N4dEQj+E60HVB9msU0KTsKOaYzkcLQvWpFYQfB4tygjPM4wtcuQflykWwQjcuzrFPmsealltmFWNzwa/42qxqHgyoHJycrq/eOvrekHrL02guCsgp2qHXrJsI0/xvxAlzc8lv+LmubVUbqWaMqOARtL00pnGhNdBTRuqn/WpL0TNOC/Wn8BT5fVrSt3n8LNbQd9EbLDZSP/6hv3SFa/mWBttvBYDkUtgaM7bv7H3Cg774pddW1RL/SXAJdb+c9RuxTpyqyPeCC26/hzugPfXB3BwTlnpbyRkMagQ8X+fmV1S92r/lNn+LpD2BdWnU=).

I guess the difference is because Vue is unwrapping refs when resolving them. Also see https://github.com/orgs/vuejs/discussions/9564#discussioncomment-7518938.
